### ISSUE TITLE
Is it expected to have to re-auth once a week?

### ISSUE THREAD

**Issue by guiniol** [gauteh/lieer#196]
Hi there!

I set up gmailieer recently, and it worked without a hitch.
I decided to use my own custom key to avoid throttling (little good that seemed to do :smile: ), but now I see that I have to re-auth once a week.
Does that also happen with the shared API key? Did I setup my private API key wrong?

Cheers,

-- 
guiniol

**Comment by guiniol** [comment]
Ack.
No idea how to fix it then.
I'll see what I find out.

**Comment by gauteh** [comment]
Do you also specify the private api key everytime you use gmi? Otherwise it
will fail to refresh token, and you need to re-auth.

man. 29. mar. 2021, 20:51 skrev guiniol ***@***.***>:

**Comment by guiniol** [comment]
I have the `-c` option with the path to my API key. Is there something else I should specify?

**Comment by gauteh** [comment]
No, that should be enough. As long as you always use it.

**Comment by guiniol** [comment]
Tried with the API key bundled with gmailieer, and I get the same results. No idea what is going on. Maybe I need to disable some kind of security setting in my account?

**Comment by guiniol** [comment]
Still happening.
I must be doing something wrong if I am the only person with this problem.
Could you share how the API access is setup (as in, which parts of the Gmail API the project is allowed to access) for the shared API key?

**Comment by gauteh** [comment]
Think that should be here: https://github.com/gauteh/lieer/blob/master/lieer/remote.py#L29

**Comment by guiniol** [comment]
Thanks, I adjusted accordingly. We'll see in a week :D

**Comment by guiniol** [comment]
Still happening...
I have setup the access for my API key the same way yours is (and gmi is working), but whether I use your API key or mine, I have to re-auth every week. I wonder if there is some kind of settings in my google account that is triggering that.
I have other projects access eg calendar in a similar way and no issues. It's really weird.

**Comment by khardix** [comment]
You are not alone in this, @guiniol – I have this problem also :slightly_smiling_face: Although I was not passing the custom credentials to all invocations – I have added that now. However, other software ([vdirsyncer](https://github.com/pimutils/vdirsyncer) to be specific) also display this issue, and I have that configured to use the credentials on every command.

Since this not seem to be gmi-specific, I tried to poke around and found out that the `oauth2client` is now deprecated in favor of [google-auth](https://github.com/googleapis/google-auth-library-python) and/or [oauthlib](https://github.com/oauthlib/oauthlib). Perhaps this is related (i.e. the oauth2client no longer getting the updates needed to access current Google APIs)?

**Comment by guiniol** [comment]
Thank you!!

I also use vdirsyncer and haven't seen (or haven't noticed) that yet.

**Comment by gauteh** [comment]
Any changes?

**Comment by guiniol** [comment]
No, still the same. I actually had to login to be able to sync and see your message.

**Comment by mjg** [comment]
I see the same effect when I use my own API keys for `mbsync` and such (different API parts, though). I think this is connected to the fact that the "app" that you register is considered experimental and would need to be approved by Google, but I might be completely wrong.
For `lieer`, I switched back to the shipped keys because of that.

**Comment by gauteh** [comment]
That is interesting! Can anyone else confirm whether using your own API keys also requires re-auth? We should add a note to the README about this in that case.

**Comment by guiniol** [comment]
I am still seeing this with my own API key.
Right now, I have reverted to using the shared API key to see if I still have to re-auth with it (as that was the case earlier, but I may have made a mistake somewhere). We'll see in a week.

**Comment by guiniol** [comment]
So, after reverting to the shared API key, I haven't had to re-auth. My guess would be something to do with keys created after a certain date having some authorisation removed.
